contracts/ZachRefund.sol

```diff
@@ -1,33 +1,45 @@
 //SPDX-License-Identifier: MIT
-pragma solidity ^0.8.4;
+pragma solidity ^0.8.19;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 
-contract ZachRefund is Ownable{
+contract ZachRefund is Ownable {
     bytes32 public immutable merkleRoot;
     IERC20 public immutable token;
 
-    constructor(bytes32 _merkleRoot, address _token){
+    mapping(address => bool) public claimed;
+
+    constructor(bytes32 _merkleRoot, address _token) {
         merkleRoot = _merkleRoot;
         token = IERC20(_token);
     }
 
-    function retrieveFundsETH(address payable to) external onlyOwner {
-        to.transfer(address(this).balance);
+    function retrieveFundsETH(address to) external onlyOwner {
+        to.call{value: address(this).balance}("");
```

Comment:
maybe add a require() that the call worked? otherwise gas estimation will likely underestimate and make tx revert (internally)

its no big deal, but caller needs to know that they have to increase gas limit manually